sqlglot/generator.py

```diff
@@ -2426,6 +2427,9 @@ def expressions(
         result_sqls = []
         for i, e in enumerate(expressions):
             sql = self.sql(e, comment=False)
+            if not sql:
```

Comment:
why was this never hit before?

Reply:
Probably didn't have a use case where an expression was generating an empty string.

Reply:
what hits it now?

Reply:
The test case I tried was actually mis-formatted due to this line: https://github.com/tobymao/sqlglot/pull/1958/files#diff-362498362ae539e216cd83ebfe510977107648742f1a1bee7b2db7b5b7d27974L992: `TemporaryProperty` generates `""` for T-SQL and so even though `expressions` is simply `[""]` in this section, we still output a redundant space separator.

I mentioned this in a [comment](https://github.com/tobymao/sqlglot/pull/1958#discussion_r1274042732) above, and you can easily see that if we had more properties in the DDL we'd hit the same issue due to the `sep.join(...)` in the `expressions` method.